### ISSUE TITLE
Unable to use application behind corporate proxy

### ISSUE THREAD

**Issue by martinskeem** [MicrosoftTranslator/DocumentTranslation#113]
HttpClient requests are caught up in our corporate proxy. Without an ability to define proxy settings it doe not work for us.

I need to be able to provide the proxy address and let HttpClient use the user's default credentials, e.g.:

```c#
var proxy = new WebProxy 
{ 
    UseDefaultCredentials = true, 
    Address = "abc"
};

var httpClientHandler = new HttpClientHandler { Proxy = proxy };
var httpClient = new HttpClient(handler: httpClientHandler, disposeHandler: true);
```

I have been searching to find a way to do this with a configuration, but without any luck. HTTP_PROXY will only get me so far - I still need to instruct HttpClient to use the user's credentials.

I have a local build where the user can define proxy settings in the Settings tab and where the HttpClients are created using a factory method reflecting the settings. I would like to open a PR with this, if you will accept that. The PR would also make changes to [DocumentTranslationService](https://github.com/MicrosoftTranslator/DocumentTranslationService). 

Illustration of Settings from my local branch:

![image](https://github.com/MicrosoftTranslator/DocumentTranslation/assets/15797029/15a856e5-7e21-466a-ba56-0b8509b0bf07)

**Comment by chriswendt1** [comment]
Please go ahead with the PR. This has a chance to address other users' observations as well.
Thank you for your contribution!

**Comment by martinskeem** [comment]
Added the PR and associated it to this issue. Let me know what you think.

**Comment by chriswendt1** [comment]
Resolved with #116 .
